### PR TITLE
Move CLI code out of `__main__.py` to allow multiprocessing to work for `python -m sphinxlint`

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/sphinx-contrib/sphinx-lint/issues/83.

We need to import the function used by multiprocessing into `__main__.py` for it to work via `python -m sphinxlint`. So let's move all of it out and keep a minimal `__main__.py`.

### CODE REVIEW

**Review by jaraco — APPROVED** [review]

File: sphinxlint/__main__.py [hunk 1]
```diff
@@ -1,246 +1,4 @@
...
-            pool.close()
-            pool.join()
-
-    return int(bool(count))
-
+from sphinxlint import cli
 
 if __name__ == "__main__":
-    sys.exit(main())
+    cli.main()
```
Comment (ezio-melotti) [inline sphinxlint/__main__.py:4]: Why was the `sys.exit` removed?

Comment (hugovk) [reply sphinxlint/__main__.py]: Mistake! Thanks, added back.

**Comment by hugovk** [comment]
I'll make a release with this.

**Comment by hugovk** [comment]
:rocket:

https://pypi.org/project/sphinx-lint/0.8.2/
https://github.com/sphinx-contrib/sphinx-lint/releases/tag/v0.8.2

**Pull request merged**